clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10340,10 +10399,10 @@ void OffloadPackager::ConstructJob(Compilation &C, const JobAction &JA,
           static_cast<const toolchains::SYCLToolChain &>(*TC);
       SYCLTC.AddImpliedTargetArgs(TC->getTriple(), Args, BuildArgs, JA, *HostTC,
                                   Arch);
-      SYCLTC.TranslateBackendTargetArgs(TC->getTriple(), Args, BuildArgs, Arch);
+      SYCLTC.TranslateBackendTargetArgs(TC->getTriple(), Args, BuildArgs);
       createArgString("compile-opts=");
       BuildArgs.clear();
-      SYCLTC.TranslateLinkerTargetArgs(TC->getTriple(), Args, BuildArgs, Arch);
+      SYCLTC.TranslateLinkerTargetArgs(TC->getTriple(), Args, BuildArgs);
```

Comment:
I removed the Arch arg here because it does not add the arguments when using `-Xsycl-target-backend=spir64_gen '-device pvc -FOO' -Xsycl-target-linker=spir64_gen -LFOO` correctly; it would only add them in a case where the `=spir64_gen` was removed or it was using the `intel_gpu_pvc` syntax.

Reply:
By arguments, are you referring to the values '-device pvc -FOO' and -LFOO, getting passed to the target backend and linker, respectively?

Reply:
Yes, I was referring to those; `-device pvc -FOO` and `-LFOO` were not appearing in `compile-opts` or `link-opts` in the `--image` argument.